### DISCUSSION TITLE
Support for OneDev Git Server

### DISCUSSION THREAD

**Discussion by CronixZero** [renovatebot/renovate#29160] [Suggest an Idea]
### Tell us more.

[Onedev](https://github.com/theonedev/onedev) is a selfhosted Git Server which supports Pull Requests, Builds, Issues and Code Repositories in one application.

Renovate is an amazing product for updating dependencies and keeping track with updates. I would really appreciate to be able to use it on OneDev, which currently hosts the Kubernetes GitOps Configuration and a lot of code of me and my team. Currently there aren't really any alternatives for keeping track of Updates on OneDev.

https://code.onedev.io/~help/api contains a Rest API Viewer where all Endpoints are listed.

Thank you!

**Comment by rarkins** [comment]
A community PR would be welcomed

**Reply by CronixZero** [reply]
Thank you for the fast reply! 
What are the main areas that would need change for this to be implemented? I already searched through the documentation but couldn't find anything related to adding new platforms.
My current understanding is that https://github.com/renovatebot/renovate/tree/main/lib/modules/platform contains all the platforms and new platforms could be added by creating a new folder, using another platform as a template and then adding the new platform to [api.ts](https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/api.ts). Is that right? And are there any other things one should think of when adding a new platform (like any other references which do not solely rely on the Platform interface as an interaction endpoint)

**Reply by rarkins** [reply]
Here's an example of someone adding a platform right now: https://github.com/renovatebot/renovate/pull/29151

**Comment by jamietanna** [comment]
Interestingly, OneDev now has support for Renovate natively (when paying for the OneDev Enteprise edition)

Related: https://xcancel.com/Foorack/status/2052075031627276538
